### PR TITLE
add ranks

### PR DESCRIPTION
**Pull request by Ama16**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/tinkoff-ai/etna-ts/blob/master/CONTRIBUTING.md)?
- [ ] Did you update the docs? We use Numpy format for all the methods and classes.
- [x] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/tinkoff-ai/etna-ts/blob/master/CHANGELOG.md)?

## Type of Change

- [ ] Examples / docs / tutorials / contributors update
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves an existing feature)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: etna/analysis/feature_relevance/relevance.py [hunk 1]
```diff
@@ -21,8 +22,15 @@ def __init__(self, greater_is_better: bool):
         """
         self.greater_is_better = greater_is_better
 
+    def _get_ranks(self, table) -> pd.DataFrame:
```
Comment (alex-hse-repository) [inline etna/analysis/feature_relevance/relevance.py:25]: Add also typing to `table` pls

File: etna/analysis/feature_relevance/relevance.py [hunk 2]
```diff
@@ -21,8 +22,15 @@ def __init__(self, greater_is_better: bool):
...
+        """Compute rank relevance table from relevance table."""
+        rank_table = table.apply(lambda x: pd.Series(scipy.stats.rankdata(x.values), index=x.index), axis=1)
```
Comment (alex-hse-repository) [inline etna/analysis/feature_relevance/relevance.py:27]:
```suggestion
       rank_table =  pd.DataFrame(scipy.stats.rankdata(rank_table, axis=1), columns= rank_table.columns, index= rank_table.index)
```

File: etna/analysis/feature_relevance/relevance.py [hunk 3]
```diff
@@ -21,8 +22,15 @@ def __init__(self, greater_is_better: bool):
...
+        if self.greater_is_better:
+            rank_table = -1 * (rank_table - rank_table.shape[1] - 1)
```
Comment (alex-hse-repository) [inline etna/analysis/feature_relevance/relevance.py:29]: Why not just `table *=-1` before ranking?

File: etna/transforms/feature_importance.py [hunk 4]
```diff
@@ -208,7 +212,9 @@ def fit(self, df: pd.DataFrame) -> "MRMRFeatureSelectionTransform":
         self.clustering.build_distance_matrix(ts=ts)
         self.clustering.build_clustering_algo(n_clusters=self.n_clusters, linkage=self.linkage)
         s2c = self.clustering.fit_predict()
-        relevance_table = self.relevance_method(ts[:, :, "target"], ts[:, :, ts.regressors], **self.relevance_params)
+        relevance_table = self.relevance_method(
```
Comment (alex-hse-repository) [inline etna/transforms/feature_importance.py:215]: It is better to use keyword arguments hear(imho)

File: tests/test_analysis/test_feature_relevance/test_relevance.py [hunk 5]
```diff
@@ -8,11 +8,21 @@ def test_statistics_relevance_table(simple_df_relevance):
...
 
 def test_model_relevance_table(simple_df_relevance):
     rt = ModelRelevanceTable()
     assert rt.greater_is_better
     df, df_exog = simple_df_relevance
-    assert rt(df=df, df_exog=df_exog, model=DecisionTreeRegressor()).shape == (2, 2)
+    assert rt(df=df, df_exog=df_exog, return_ranks=False, model=DecisionTreeRegressor()).shape == (2, 2)
+
+
+def test_relevance_table_ranks(simple_df_relevance):
```
Comment (alex-hse-repository) [inline tests/test_analysis/test_feature_relevance/test_relevance.py:21]: Better to make a fixture for the table with ranks and test the method in two cases of `greater_is_better`

File: tests/test_transforms/test_feature_importance_transform.py [hunk 6]
```diff
@@ -73,7 +73,7 @@ def ts_with_regressors():
 def test_mrmr_right_len(relevance_method, clustering_method, top_k, ts_with_regressors):
```
Comment (alex-hse-repository) [inline tests/test_transforms/test_feature_importance_transform.py:73]: Better to use keyword arguments, fix it pls in this file

**Review by alex-hse-repository — CHANGES_REQUESTED** [review]

File: etna/analysis/feature_relevance/relevance.py [hunk 7]
```diff
@@ -59,7 +71,9 @@ class ModelRelevanceTable(RelevanceTable):
     def __init__(self):
         super().__init__(greater_is_better=True)
 
-    def __call__(self, df: pd.DataFrame, df_exog: pd.DataFrame, **kwargs) -> pd.DataFrame:
+    def __call__(self, df: pd.DataFrame, df_exog: pd.DataFrame, return_ranks: bool, **kwargs) -> pd.DataFrame:
```
Comment (julia-shenshina) [inline etna/analysis/feature_relevance/relevance.py:74]: shouldn't we add `False` as default value for `return_ranks`?

**Review by julia-shenshina — DISMISSED** [review]

**Review by alex-hse-repository — DISMISSED** [review]

**Review by alex-hse-repository — APPROVED** [review]

**Pull request merged**